src/FsMinifilter/Main.cpp

```diff
@@ -0,0 +1,73 @@
+#include "FsMinifilter.h"
+
+//
+// The minifilter handle that results from a call to FltRegisterFilter
+// NOTE: this handle must be passed to FltUnregisterFilter during minifilter unloading
+//
+PFLT_FILTER g_minifilterHandle = NULL;
+
+//
+// Constant FLT_REGISTRATION structure for our filter. This
+// initializes the callback routines our filter wants to register for. 
+// This is only used to register with the filter manager
+//
+CONST FLT_OPERATION_REGISTRATION g_callbacks[] =
+{
+    {
+        IRP_MJ_CREATE,
+        0,
+        PreOperationCreate,
+        0
+    },
+
+    { IRP_MJ_OPERATION_END }
+};
+
+//
+// The FLT_REGISTRATION structure is used to provide information about
+// a file system minifilter to the filter manager.
+//
+CONST FLT_REGISTRATION g_filterRegistration =
```

Comment:
`g_filterRegistration` and `g_callbacks` can be local, no need to make them global.